vue.config.js

```diff
@@ -0,0 +1,60 @@
+const path = require('path')
```

Comment:
The structure I got from template v5.0.8 looks a bit different.
```
const { defineConfig } = require('@vue/cli-service')
module.exports = defineConfig({
  transpileDependencies: true,
  lintOnSave: false
})
```

First of all it does not assign the configuration object to `module.exports` directly but invokes `defineConfig` on it.
Secondly it sets `transpileDependencies: true`, which I cannot see in here.

Reply:
O.K., this one is the hardest to explain and I had to do some tests to remember where this all comes from.

Concerning the call through `defineConfig` first, this is something new added in `Vue-CLI v5` which is kind of optional as stated in [the official documentation](https://cli.vuejs.org/config/#vue-config-js).
You can see that we can use two different formats and that I used the first one instead of the second. In the source code, we can see that [defineConfig](https://github.com/vuejs/vue-cli/blob/v5.0.8/packages/%40vue/cli-service/lib/Service.js#L375:L376) doesn't do anything else than type hint the content of the object inside, which is also done explicitly in the first version.
The official documentation states that `defineConfig could provide better intellisense support`. However, if you look [in the source once again](https://github.com/vuejs/vue-cli/blob/v5.0.8/packages/%40vue/cli-service/types/index.d.ts#L136:L138), this would only be relevant if a function is used instead of an object which is not our case.
In the rest of the documentation, the good old `module.exports` is used in all examples. So at that time, I decided to stick to the first format which is IMHO much clear and transparent compared to calling a function burried deep inside the code and which is not well documented.
Moreover, as this `defineConfig` isn't used anywhere else in the official documentation itself, I personally think it makes you wonder why the explanations never use it and why you have to write your config file differently than what the official documentation explains. However, this can easily be changed if you want it though...

`transpileDependencies: true` is linked to the same documentation discrepancy thing but has a more deep impact that we can argue here:
This parameter is linked to which modules from `node_modules` should be transpiled through `babel-loader` and which shouldn't and should be kept as is. I think the two major which concerns us here is `Vuetify` and `OpenLayers`.
First, the template sets it to true as you mentioned up here but the [official documentation](https://cli.vuejs.org/config/#transpiledependencies) says this defaults to `false` if not specified and that `transpiling all the dependencies could slow down the build process` and let you specify an array of dependencies to transpile.
When I first made the changes for this PR in first half of 2022, I remember that dependencies installed at that time made errors while transpiling `Vuetify` so I had to remove this line. Today it's not the case anymore, only warnings can be seen, so it's true that we could potentially put this line back to stick to the official template... which diverges from the official documentation.

When generating a new `Vuetify` app instead of a pure `Vue` one, this line is changed from `transpileDependencies: true` to `transpileDependencies: ['vuetify']`. So I thought about looking at this when including the `Vuetify` plugins and `ESLint` support...

However there's another point to think about beforehand today:
Before Vue-CLI v5, when building your project, it was passed through `Babel` normally, but you could also have your app built in [modern mode](https://cli.vuejs.org/guide/browser-compatibility.html#modern-mode). There, `Vue CLI` would produce two versions of the app, one legacy bundle transpiled to `ES5` and one modern bundle aiming `ES2015+` code, both added in the generated `html` files. With a simple technique, modern browsers are able to skip the legacy bundle and load only the modern one.
`Vue-CLI v5`, [has made the modern mode the default one](https://cli.vuejs.org/migrations/migrate-from-v4.html#the-build-command-and-modern-mode). And that's where funny things happen, when `npm run build` is launched today, with its new way of working which takes into account the content of `browserlist`, we can see those two lines appear:

> All browser targets in the browserslist configuration have supported ES module.
> Therefore we don't build two separate bundles for differential loading.

It means the generated app is now only available to `ES2015+` compatible browsers. We can still transpile some dependencies so that potential polyfills for `ES2022` or `ES2023` novelties are added, but this won't have the same impact as before where all was transpiled to `ES5`.
Note also that we can force the generation of a legacy build by passing a `--legacy` flag to the build command.

[`Vuetify` official documentation](https://v2.vuetifyjs.com/en/getting-started/browser-support/#browsers) states that only `IE11` and `Safari 9` would require the use of polyfills, both of which are not taken into account with current `browserlist` file. So it means adding `Vuetify` in `transpileDependencies` changes nothing in terms of compatibility with current configuration, except we can see some warnings telling us transpiling makes worse than better:

> [BABEL] Note: The code generator has deoptimised the styling of C:\Sources\Wegue\UpdateCLI\node_modules\vuetify\dist\vuetify.js as it exceeds the max of 500KB.
> [BABEL] Note: The code generator has deoptimised the styling of C:\Sources\Wegue\UpdateCLI\node_modules\@mapbox\mapbox-gl-style-spec\dist\index.es.js as it exceeds the max of 500KB.

I haven't looked for `OpenLayers` though, knowing that the current version has certainly not the same needs as the one we're trying to update to in #320 ...

So to summarize, we can argue here with such questions:

- Should we modify the `browserlist` file to target a broader range of browsers and more polyfills are added while bundling or should we stick with the current content ?
- Should we prefer to set `transpileDependencies: true` which has no effect in terms of compatibility with current `browserlist` configuration (except perhaps for `OpenLayers`, to be checked depending on the version we're aiming for) but warns that styles are not optimized anymore or should we aim for maximum compatibility ?
- Should we aim only modern browsers with the release of `Wegue V2` and so remove this `transpileDependencies` by default, setting a note in the documentation about this and explaining to users what should be done in case of compatibility issues ?

I don't know what you and @chrismayer think about this, that's only a single line to change in the end in terms of code, but gives a very different message to the people who will use `Wegue V2`...

Sorry for the waaaaay tooooo long comment once again and hope this is all understandable...

Cheers,
Sébastien

Reply:
Thanks a lot Sebastien for the very conclusive explanation.

I myself tumbled over the `browserlist` build target console print, which told me there's currently only output for the 'ES2015+' target. It`s a good question, whether we should support legacy browsers. My personal opinion is, since this hasn't been supported previously we could just leave it as is. If it is required then we could address that later on in a separate PR.

As you proposed, I would just stick with the structure of the template here.
* Invoke `defineConfig` - even though it's a no-op.
* Then set `transpileDependencies: false` explicitly , maybe with a short comment for the reasoning as pointed out above and that it could be granularly enabled. I totally agree with you, that we should not impose a compile overhead, when there is absolutely no need to.

This way it's immediately apparent where we are going down our own track, just in case we have to pinpoint that later.
Note: There is no problem related to OL when not transpiling dependencies, more later.